frontend/src/pages/CourseSelector/CourseSearchBar/CourseSearchBar.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { Select, Spin } from 'antd';
+import axios from 'axios';
+import { SearchCourse } from 'types/api';
+import { useDebounce } from 'use-debounce';
+import prepareUserPayload from 'utils/prepareUserPayload';
+import type { RootState } from 'config/store';
+import { addTab } from 'reducers/courseTabsSlice';
+
+const CourseSearchBar = () => {
+  const [value, setValue] = useState<string | null>(null);
+  const [courses, setCourses] = useState<{ label: string, value: string }[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const [debouncedSearchTerm] = useDebounce(value, 200);
+
+  const dispatch = useDispatch();
+
+  const planner = useSelector((state: RootState) => state.planner);
+  const degree = useSelector((state: RootState) => state.degree);
+
+  useEffect(() => {
+    const searchCourse = async (query: string) => {
+      try {
+        const res = await axios.post<SearchCourse>(
+          `/courses/searchCourse/${query}`,
+          JSON.stringify(prepareUserPayload(degree, planner)),
+        );
+        setCourses(
+          Object.keys(res.data).map((course) => ({
+            label: `${course}: ${res.data[course]}`,
+            value: course,
+          })),
+        );
+      } catch (err) {
+        // eslint-disable-next-line
+        console.log(err);
```

Comment:
if we have established this pattern, might as well:
```suggestion
        console.log("error at CourseSearchBar", err);
```

Reply:
Ooops must've missed this. cheers for the pickup!